### PR TITLE
feat(react-components): Add NextClippingCommand to the clip tool

### PR DESCRIPTION
**Pull request by nilscognite**
#### Type of change

Feat  

#### Jira ticket :blue_book:
https://cognitedata.atlassian.net/browse/BND3D-4442

## Description :pencil:

This takt the next copping box or slicing planes (all slicing planes is regarded as a group) and use it as clipping.

## How has this been tested? :mag:

Open storybook/architecture
Set Crop tool active
Make som boxes and slices
Click at crop
Click at 'Set the next crop box or slicing plane as global clipping' some times.

## Checklist :ballot_box_with_check:

- [x] I am proud of this feature.
- [x] I have performed a self-review of my own code.
- [x] I have added PR type (Feat, Bug, Chore, Test, Docs, Style, Refactor) to the PR title.
- [x] I have manually tested this for different scenarios (different models, formats, devices, browsers).
- [x] I have commented my code in hard-to-understand areas.
- [ ] I have made corresponding changes to the public documentation.
- [ ] I have added unit and visuals tests to prove that my feature works to the best of my ability.
- [ ] I have refactored the code for readability to the best of my ability.
- [x] I have checked that my changes do not introduce regressions in the public documentation.
- [ ] I have outlined any known defects / lacking capabilities in the contents of this PR.
- [ ] I have listed any remaining work that should follow this PR in the description or jira/miro/etc.
- [ ] I have added TSDoc to any public facing changes.
- [ ] I have added "developer documentation" in any package README.md that is related / applicable to this PR.
- [ ] I have noted down and am currently tracking any technical debt introduced in this PR.
- [x] I have thoroughly thought about the architecture of this implementation.
- [ ] I have asked peers to test this feature.

### CODE REVIEW

File: react-components/src/architecture/concrete/clipping/commands/NextClippingCommand.ts [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        count++;
+      }
+    }
+    if (this.rootDomainObject.getDescendantByType(SliceDomainObject) !== undefined) {
+      count++;
+    }
+    return count >= 2;
+  }
+
+  protected override invokeCore(): boolean {
```
Comment (pramodcog) [inline react-components/src/architecture/concrete/clipping/commands/NextClippingCommand.ts:45]: There are lots going on in this function, would be nice to separate it with helper functions like
```
getCropBoxesAndSlices()
getSelectedIndex()
and others()
```

**Comment by greenbech** [comment]
I think I've found a bug. It's probably easiest to watch the screen recording. It seems like the `"Show or hide all other slicing planes and crop boxes"` gets into a weird state when you use it together with the next crop box button. Only when you click on the first selected crop box do the other crop boxes show up. 

https://github.com/cognitedata/reveal/assets/7979907/8bd4b3e0-1d76-4f68-b953-bfd54c9a69d9

**Comment by nilscognite** [comment]
> I think I've found a bug. It's probably easiest to watch the screen recording. It seems like the `"Show or hide all other slicing planes and crop boxes"` gets into a weird state when you use it together with the next crop box button. Only when you click on the first selected crop box do the other crop boxes show up.
> 
>  Screen.Recording.2024-06-25.at.14.00.29.mov

This is now fixed. I forgot the set them visible(invisible

File: react-components/src/architecture/concrete/clipping/commands/NextClippingCommand.ts (hunk 1 above)
Comment (nilscognite) [reply react-components/src/architecture/concrete/clipping/commands/NextClippingCommand.ts:71]: This is now fixed in 10e758685d41e3b28aef8d9e9745f874bd90ae40

**Review by pramodcog — APPROVED** [review]

**Pull request merged**